test-case-reporting/test/test_result_record.test.ts

```diff
@@ -0,0 +1,148 @@
+    }),
+}));
+
+describe('TestResultRecord', () => {
+  let db: Database;
+  let testResultRecord: TestResultRecord;
+  let sampleBuild: Travis.Build;
+  let sampleJob: Travis.Job;
+  let sampleKarmaReport: KarmaReporter.TestResultReport;
+
+  beforeAll(async () => {
+    db = dbConnect();
+    await setupDb(db);
+
+    testResultRecord = new TestResultRecord(db);
+
+    sampleBuild = (getFixture(
+      'sample-travis-build'
+    ) as unknown) as Travis.Build;
+
+    sampleJob = (getFixture('sample-travis-job') as unknown) as Travis.Job;
+
+    sampleKarmaReport = (getFixture(
+      'sample-karma-report'
+    ) as unknown) as KarmaReporter.TestResultReport;
+  });
+
+  afterAll(() => {
+    db.destroy();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    db('test_runs').truncate();
+    db('test_cases').truncate();
+    db('jobs').truncate();
+    db('builds').truncate();
+  });
+
+  describe('insertTravisBuild', () => {
+    it('adds the build to the database', async () => {
+      await testResultRecord.insertTravisBuild(sampleBuild);
+
+      const build = await db<DB.Build>('builds').select().first();
+      expect(build.commit_sha).toEqual(sampleBuild.commitSha);
+      expect(build.build_number).toEqual(sampleBuild.buildNumber);
+    });
+  });
+
+  describe('insertTravisJob', () => {
+    let buildId: number;
+
+    beforeEach(async () => {
+      buildId = await testResultRecord.insertTravisBuild(sampleBuild);
+    });
+
+    it('adds the job to the database if build id is good', () => {
+      testResultRecord.insertTravisJob(
+        {
+          jobNumber: '413413.612',
+          testSuiteType: 'unit',
+        },
+        buildId
+      );
+    });
+
+    it('fails to add the job to the database if build id is bad');
+  });
+
+  describe('testStatus', () => {
+    it('returns SKIP when the test was skipped', () => {
+      expect(testResultRecord.testStatus(true, false)).toEqual('SKIP');
+    });
+
+    it('returns PASS when the test passed', () => {
+      expect(testResultRecord.testStatus(false, true)).toEqual('PASS');
+    });
+
+    it('returns FAIL when the test failed', () => {
+      expect(testResultRecord.testStatus(false, true)).toEqual('FAIL');
+    });
+  });
+
+  describe('storeTravisResults', () => {
+    it('inserts test cases', async () => {
+      testResultRecord.storeTravisReport({
+        job: sampleJob,
+        build: sampleBuild,
+        result: sampleKarmaReport,
+      });
+
+      const testCases: Array<DB.TestCase> = await db<DB.TestCase>(
+        'test_cases'
+      ).select();
+
+      const sampleTestCases: Array<DB.TestCase> = sampleKarmaReport.browsers.results.map(
+        result => {
+          const {description, suite} = result;
+          const name = testResultRecord.makeTestCaseName(description, suite);
+          return {
+            id: md5(name),
+            name,
+          };
+        }
+      );
+
+      expect(sampleTestCases).toMatchObject(testCases);
```

Comment:
So in this case, trying to generate the expected result ends up with you largely re-implementing the method. If there's a mis-match, there'a good chance it's because of some bug in the test code generating the expected result. Instead, prefer to hand-write what the expected result is (this can also apply above where you check against `sampleBuild` instead of the actual values from the fixture). This will help ensure that bugs in testing code don't mask bugs in real code. being explicit with testing code can be preferable, even when it means the tests can get a bit verbose.

If the verbosity seems excessive, you can always test partially, like by checking the length of the test cases list or just the `(status, name)` tuples and verifying they are as expected.

Reply:
Done.